### PR TITLE
feat(anvil): `--disable-tracing`

### LINKED ISSUE #7442: Add CLI Argument to Disable Logging on Anvil
### Component

Anvil

### Describe the feature you would like

It would be great to have a CLI argument that we could use to disable the logging on calls to the Anvil RPC, particularly for workloads which involve lots of RPC calls as this may be more performant.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #7442 

## Solution

Add CLI arg `--disable-tracing` to disable all tx tracing in anvil. Default value is `false`.

### CODE REVIEW

File: crates/anvil/src/config.rs [hunk 1]
```diff
@@ -813,6 +816,14 @@ impl NodeConfig {
         self
     }
 
+    /// Sets whether to disable tracing
+    #[must_use]
+    pub fn with_disable_tracing(mut self, yes: bool) -> Self {
+        println!("Disabling tracing {:?}", yes);
```
Comment (mattsse) [inline crates/anvil/src/config.rs:822]:
```suggestion

```

File: crates/anvil/src/eth/backend/executor.rs [hunk 2]
```diff
@@ -284,9 +285,12 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
...
         // records all call and step traces
-        let mut inspector = Inspector::default().with_tracing();
-        if self.enable_steps_tracing {
-            inspector = inspector.with_steps_tracing();
+        let mut inspector = Inspector::default();
+        if !self.disable_tracing {
+            inspector = inspector.with_tracing();
+            if self.enable_steps_tracing {
+                inspector = inspector.with_steps_tracing();
+            }
```
Comment (mattsse) [inline crates/anvil/src/eth/backend/executor.rs:293]: we can leave the setup as is, but if tracing is disabled we need to change this:

foundry_evm::utils::new_evm_with_inspector(&mut *self.db, env, &mut inspector);

to evm setup without inspector, which is always faster than with any inspector

**Review by mattsse — CHANGES_REQUESTED** [review]
if we disable tracing we should skip the inspector entirely

File: crates/anvil/src/eth/backend/executor.rs [hunk 3]
```diff
@@ -284,15 +286,49 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
...
+            // transact and commit the transaction
+            match evm.transact_commit() {
+                Ok(exec_result) => exec_result,
+                Err(err) => {
+                    warn!(target: "backend", "[{:?}] failed to execute: {:?}", transaction.hash(), err);
+                    match err {
+                        EVMError::Database(err) => {
+                            return Some(TransactionExecutionOutcome::DatabaseError(
+                                transaction,
+                                err,
```
Comment (mattsse) [inline crates/anvil/src/eth/backend/executor.rs:311]: I think we don't need the duplication here if we make the if else block return the result?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/anvil/src/eth/backend/executor.rs (hunk 3 above)
Comment (yash-atreya) [reply crates/anvil/src/eth/backend/executor.rs:311]: The issue was building the `evm` with different types, one with an inspector and one without.

**Comment by zerosnacks** [comment]
Would be great to get this updated / merged @yash-atreya when you have opportunity

**Comment by yash-atreya** [comment]
Closing in favour of the `--silent` flag

**Pull request closed without merging**